src/ICeWall/doc/README.md

```diff
@@ -0,0 +1,97 @@
+
+The connect_by_abutment key defines the list of pins on the edges of the IO cells that are connected by abutment forming a ring of these signals around the die. ICeWall will connect these abuted signals together into SPECIAL netst, so that the router will not try to add wiring for these pins.
+
+The pad_pin_name key defines the name of the pin of the IO cell that is connected to the chip package.
+
+The pad_pin_layer key defines the layer that will be used to create a physical pin on the die that is coincident with the location of the external connection to the chip package
+
+The breakers key defines the list of cell types in the types definition that act as signal breakers for the signals that connect by abutment.
+
+The cells key defines additional data for each IO cell. Each type defined in the types key must have a matching definition in the list of cells. Within a cell definition, the following keys may be used
+
+* cell__name
+* orient
+* physical_only
+* breaks
+* signal_type 
+* power_pad
+* ground_pad
+* pad_pin_name
+
+| Key | Description |
+| --- | --- |
+| cell_name | Can either be the name of the LEF macro to use, or else a dictionary allowing a different cell name to be specified for each side of the die. Sides are denoted by bottom right top and left |
+| orient | A dictionary that specifies the orientation of the cell for each side of the die |
+| physical_only | Cells that are not associated with cells in the design, primarily used for physical connections through the pad ring |
+| breaks | For breaker cells, defines the abutment signals which are broken by the presence of this cell. Each signal broken requires a list of pins on the breaker cell for the left and right hand side of the cell (An empty list may be provided if there are no pins for the broken signals |
+| signal_type | Set to 1 if this cell is to be regarded as a signal type |
+| power_pad | Optional. Set to 1 to define this cell as a power pad |
+| ground_pad | Optional. Set to 1 o define this cell as a ground pad |
+| pad_pin_name | Optional. Define the name of the pin on the macro that connects to the chip package |
+
+
+### TCL Command Reference
+
+Details on the [TCL command interface](TCL_Interface.md)
+
+
+### Package data
+
+As an alternative to using TCL commands, the definition of the padring can be done using a padring strategy file, which allows for batch loading of the padcell data all at once.
+
+For an example of what a padring strategy file looks like refer to [../test/soc_bsg_black_parrot_nangate45/bsg_black_parrot.package.strategy](../test/soc_bsg_black_parrot_nangate45/bsg_black_parrot.package.strategy).
+
+### Signal mapping
+
+Assignment of signals to padcells can be done via a signal mapping file. This file consists of multiple lines, each one mapping a named padcell to a signal in the design. 
```

Comment:
hmm. mapping file? There really shouldn't be external files. We finally got rid of the TR config file. I find the whole architecture of the PDN/pad code to be inconsistent with the concept of a command based tool. I realize this isn't the place to fix it, but I hate to see continued investment in this methodology.

Reply:
The TCL API that is added by this PR means that you can perform the signal mapping through TCL commands, so we could entertain the thought of deprecating the external files that are currently being used.

Reply:
@Colin-Holehouse in a similar vein would it be possible to add TCL commands to handle the library setup file too? If that's possible, that way ICeWall would no longer be dependent on external files and everything could be done from the command line.

Reply:
Yes, this is the next logical step down this path.

Reply:
No side files needed anymore